nixos/tests/nvidia-container-toolkit.nix

```diff
@@ -0,0 +1,177 @@
+              if [[ $(${lib.getExe pkgs.file} "$file" | ${lib.getExe pkgs.gnugrep} ELF) ]]; then
+                check_file_referential_integrity "$file" || exit 1
+              else
+                echo "skipping $file: not an ELF file"
+              fi
+            done
+          }
+
+          check_directory_referential_integrity "/usr/bin" || exit 1
+          check_directory_referential_integrity "${pkgs.addDriverRunpath.driverLink}" || exit 1
+          check_directory_referential_integrity "/usr/local/nvidia" || exit 1
+        '';
+      in
+      pkgs.dockerTools.buildImage {
+        name = "cdi-test";
+        tag = "latest";
+        config = {
+          Cmd = [ "${testCDIScript}/bin/test-cdi" ];
+        };
+        copyToRoot = (
+          with pkgs.dockerTools;
+          [
+            usrBinEnv
+            binSh
+          ]
+        );
+      };
+    emptyCDISpec = ''
+      #! ${pkgs.runtimeShell}
+      cat <<CDI_DOCUMENT
+        {
+          "cdiVersion": "0.5.0",
+          "kind": "nvidia.com/gpu",
+          "devices": [
+            {
+              "name": "all",
+              "containerEdits": {
+                "deviceNodes": [
+                  {
+                    "path": "/dev/urandom"
+                  }
+                ],
+                "hooks": [],
+                "mounts": []
+              }
+            }
+          ],
+          "containerEdits": {
+            "deviceNodes": [],
+            "hooks": [],
+            "mounts": []
+          }
+        }
+      CDI_DOCUMENT
+    '';
+    nvidia-container-toolkit = {
+      enable = true;
+      package = pkgs.stdenv.mkDerivation {
+        name = "nvidia-ctk-dummy";
+        version = "1.0.0";
+        dontUnpack = true;
+        dontBuild = true;
+        installPhase = ''
+          mkdir -p $out/bin
+          cat <<EOF > $out/bin/nvidia-ctk
+            ${emptyCDISpec}
+          EOF
+          chmod +x $out/bin/nvidia-ctk
+        '';
+      };
+    };
+  in
+  {
+    name = "nvidia-container-toolkit";
+    meta = with lib.maintainers; {
+      maintainers = [ ereslibre ];
+    };
+    nodes = {
+      no-nvidia-gpus =
+        { config, ... }:
+        {
+          environment.systemPackages = with pkgs; [ jq ];
+          hardware = {
+            inherit nvidia-container-toolkit;
+            nvidia = {
+              open = true;
+              package = config.boot.kernelPackages.nvidiaPackages.stable.open;
+            };
+          };
+        };
+
+      nvidia-one-gpu =
+        { config, pkgs, ... }:
+        {
+          virtualisation.diskSize = 10240;
+          environment.systemPackages = with pkgs; [
+            jq
+            podman
+          ];
+          hardware = {
+            inherit nvidia-container-toolkit;
+            nvidia = {
+              open = true;
+              package = config.boot.kernelPackages.nvidiaPackages.stable.open;
+            };
+            opengl.enable = true;
```

Comment:
I don't understand why Ofborg isn't complaining, we have a `mkRenamedModule` for this one